### PR TITLE
bugfix : Eth-Consensus-Version header on response header

### LINKED ISSUE #12580: eth-consensus-version not provides for JSON blocks on REST API
# 🐞 Bug Report

### Description

Calling `/eth/v2/beacon/blocks/head` should provide a header `Eth-Consensus-Version` as per https://ethereum.github.io/beacon-APIs/#/Beacon/getBlockV2 however this is not supplied if the return data is JSON.

### Has this worked before in a previous version?

Unknown.

## 🔬 Minimal Reproduction

```
curl -o /dev/null -s -H 'Accept: application/json' -w "%header{Eth-Consensus-Version}\n" http://localhost:3500/eth/v2/beacon/blocks/head

curl -o /dev/null -s -H 'Accept: application/octet-stream' -w "%header{Eth-Consensus-Version}\n" http://localhost:3500/eth/v2/beacon/blocks/head
capella
```

## 🌍  Your Environment

<pre>
  <code>
beacon-chain --version
beacon-chain version Prysm/v4.0.6/a75e78ddb42a8142763552c3799097c20c02e5d0. Built at: 2023-06-15 15:44:06+00:00
  </code>
</pre>

### PR DESCRIPTION
**Pull request by james-prysm**
**What type of PR is this?**
Bug fix

**What does this PR do? Why is it needed?**

adds the corrected header for get block and get blinded block calls.

**Which issues(s) does this PR fix?**

Fixes #12580

**Other notes for review**

### CODE REVIEW

File: beacon-chain/rpc/eth/beacon/blocks.go [hunk 1]
```diff
@@ -424,6 +426,9 @@ func (bs *Server) GetBlockV2(ctx context.Context, req *ethpbv2.BlockRequestV2) (
 	if !errors.Is(err, consensus_types.ErrUnsupportedField) {
 		return nil, status.Errorf(codes.Internal, "Could not get signed beacon block: %v", err)
 	}
+	if err := grpc.SetHeader(ctx, metadata.Pairs("Eth-Consensus-Version", version.String(blk.Version()))); err != nil {
```
Comment (rkapka) [inline beacon-chain/rpc/eth/beacon/blocks.go:429]:
```suggestion
	if err := grpc.SetHeader(ctx, metadata.Pairs(versionHeader, version.String(blk.Version()))); err != nil {
```

File: beacon-chain/rpc/eth/beacon/blocks.go [hunk 2]
```diff
@@ -424,6 +426,9 @@ func (bs *Server) GetBlockV2(ctx context.Context, req *ethpbv2.BlockRequestV2) (
...
+		return nil, status.Errorf(codes.Internal, "Could not set Eth-Consensus-Version header: %v", err)
```
Comment (rkapka) [inline beacon-chain/rpc/eth/beacon/blocks.go:430]:
```suggestion
		return nil, status.Errorf(codes.Internal, "Could not set "+versionHeader+" header: %v", err)
```

File: beacon-chain/rpc/eth/beacon/blinded_blocks.go [hunk 3]
```diff
@@ -38,7 +40,9 @@ func (bs *Server) GetBlindedBlock(ctx context.Context, req *ethpbv1.BlockRequest
 	if err != nil {
 		return nil, errors.Wrapf(err, "could not get block root")
 	}
-
+	if err := grpc.SetHeader(ctx, metadata.Pairs("Eth-Consensus-Version", version.String(blk.Version()))); err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not set Eth-Consensus-Version header: %v", err)
```
Comment (rkapka) [inline beacon-chain/rpc/eth/beacon/blinded_blocks.go:44]:
```suggestion
		return nil, status.Errorf(codes.Internal, "Could not set "+versionHeader+" header: %v", err)
```

File: beacon-chain/rpc/eth/beacon/blinded_blocks.go [hunk 4]
```diff
@@ -38,7 +40,9 @@ func (bs *Server) GetBlindedBlock(ctx context.Context, req *ethpbv1.BlockRequest
 	if err != nil {
 		return nil, errors.Wrapf(err, "could not get block root")
 	}
-
+	if err := grpc.SetHeader(ctx, metadata.Pairs("Eth-Consensus-Version", version.String(blk.Version()))); err != nil {
```
Comment (rkapka) [inline beacon-chain/rpc/eth/beacon/blinded_blocks.go:43]:
```suggestion
	if err := grpc.SetHeader(ctx, metadata.Pairs(versionHeader, version.String(blk.Version()))); err != nil {
```

File: api/gateway/apimiddleware/process_request.go [hunk 5]
```diff
@@ -116,7 +116,11 @@ func HandleGrpcResponseError(errJson ErrorJson, resp *http.Response, respBody []
 		// Something went wrong, but the request completed, meaning we can write headers and the error message.
 		for h, vs := range resp.Header {
 			for _, v := range vs {
-				w.Header().Set(h, v)
+				if strings.HasSuffix(h, "Eth-Consensus-Version") {
+					w.Header().Set("Eth-Consensus-Version", vs[0])
```
Comment (prestonvanloon) [inline api/gateway/apimiddleware/process_request.go:120]: Why does this use `vs[0]`?

Comment (james-prysm) [reply api/gateway/apimiddleware/process_request.go:120]: yeah I think that's wrong will fix

File: api/gateway/apimiddleware/process_request.go [hunk 6]
```diff
@@ -116,7 +116,11 @@ func HandleGrpcResponseError(errJson ErrorJson, resp *http.Response, respBody []
 		// Something went wrong, but the request completed, meaning we can write headers and the error message.
 		for h, vs := range resp.Header {
 			for _, v := range vs {
-				w.Header().Set(h, v)
+				if strings.HasSuffix(h, "Eth-Consensus-Version") {
```
Comment (rauljordan) [inline api/gateway/apimiddleware/process_request.go:119]: Let's use the variable instead of hardcoding the string here and also below

Comment (james-prysm) [reply api/gateway/apimiddleware/process_request.go:119]: maybe I should just extract the constants as header info somewhere else

**Review by prestonvanloon — APPROVED** [review]
LGTM - please address Raul's comment before merging

**Review by rauljordan — APPROVED** [review]

**Pull request merged**